Weather: migrate get_forecasts to a dedicated action page (#46229)
home-assistant/home-assistant.github.io: source/_actions/weather.get_forecasts.markdown

This action returns its result in a response variable, which you can use in later steps of the same automation or script.

{% include actions/ui_header.md %}

To get weather forecasts from an automation or a script:

1. Go to {% my automations title="**Settings** > **Automations & scenes**" %}.
2. Open an existing automation or script, or select **Create automation** > **Create new automation**.
3. If you're setting up a new automation, add a trigger in the **When** section. Scripts don't need a trigger. They run when something else calls them.
4. In the **Then do** section, select **Add action**.
5. Select what you want to control. Under **By target** (see [Targets](#targets)), select the weather entities you want to read.
6. From the actions shown for that target, select **Get weather forecasts**.
7. Select the **Forecast type** to retrieve.
8. Select **Save**.

### Options in the UI

{% options_ui %}
Forecast type:
  description: The scope of the forecast to retrieve. One of daily, hourly, or twice daily. When not set, defaults to daily.

>   required: false

Reviewer: ```suggestion
  required: false
{% endoptions_ui %}
```

---
title: "Get weather forecasts"
action: weather.get_forecasts
domain: weather
description: "Retrieves the forecasts from one or more weather entities."
---

Use this action to retrieve the forecast from one or more weather entities, for example to read out tomorrow's weather or to drive an automation based on the chance of rain.

This action returns its result in a response variable, which you can use in later steps of the same automation or script.

{% include actions/ui_header.md %}

To get weather forecasts from an automation or a script:

1. Go to {% my automations title="**Settings** > **Automations & scenes**" %}.
2. Open an existing automation or script, or select **Create automation** > **Create new automation**.
3. If you're setting up a new automation, add a trigger in the **When** section. Scripts don't need a trigger. They run when something else calls them.
4. In the **Then do** section, select **Add action**.

> 5. Select what you want to control. Under **By target** (see [Targets](#targets)), select the weather entities you want to read.

Reviewer: ```suggestion
5. Select what you want to control. Under **By target** (see [Targets](#targets)), pick the weather entity you want to read. You can also select an area, a device, or a label.
```

6. From the actions shown for that target, select **Get weather forecasts**.
7. Select the **Forecast type** to retrieve.

> 8. Select **Save**.

Reviewer: ```suggestion
8. In the **Response variable** field, enter a name to store the forecast data in. Choose a short, descriptive name that reflects what it holds, such as `berlin_forecast`. You'll use this name to read the forecast in later steps.
9. Select **Save**.
```
